include/common.h

```diff
@@ -1,20 +1,19 @@
-#include <cheri/cheri.h>
-#include <cheri/cheric.h>
+#include <cheriintrin.h>
 #include <stdbool.h>
 #include <stdint.h>
 #include <stdio.h>
 
 void inspect_pointer(void *ptr)
```

Comment:
What does this function do?

Reply:
Ah, it's really a pretty-printing function. Let's document that. And, in its own commit, we could consider renaming it to something like `print_cheri_permissions(void *ptr)` as that seems a better match.

Reply:
Well, it prints not only the permissions...

Reply:
Still, "inspect" is a pretty vague name :) I know it's not part of this PR, but maybe you can do another PR that renames this function to something more informative? I tend to use https://tratt.net/laurie/src/srep/ for this sort of thing (something like `find . -name "*.[ch]" | xargs srep -m -s inspect_pointer newname` tends to do the trick for me).

Reply:
OK, this will be another PR. Nice tool, BTW :)